File: ports/csharp/src/gui/layout/BcsBoxLayout.cs
namespace BCS.Gui.Layout {
    public enum BcsDirection {
        Horizontal,
        Vertical
    }

    // BcsBoxLayout organizes widgets into a horizontal or vertical array
    public class BcsBoxLayout : BcsLayout {
        public BcsDirection Direction { get; set; }

        public BcsBoxLayout(BcsDirection direction) {
            Direction = direction;
        }

        // Invalidate triggers an algorithmic recalculation of widget bounds
        public void Invalidate() {
            // Recompute sizes
        }
    }
}

File: ports/java/src/gui/layout/BcsBoxLayout.java
package bcs.gui.layout;

// BcsBoxLayout organizes widgets into a horizontal or vertical array
public class BcsBoxLayout extends BcsLayout {
    public enum BcsDirection {
        HORIZONTAL,
        VERTICAL
    }

    public BcsDirection direction;

    public BcsBoxLayout(BcsDirection direction) {
        this.direction = direction;
    }

    // Invalidate triggers an algorithmic recalculation of widget bounds
    public void invalidate() {
        // Recompute sizes
    }
}
